test/eredis_tests.erl

```diff
@@ -10,10 +10,14 @@ multibulk_test_() ->
                    list_to_binary(create_multibulk(["SET", "foo", "bar"]))),
      ?_assertEqual(<<"*3\r\n$3\r\nSET\r\n$3\r\nfoo\r\n$3\r\nbar\r\n">>,
                    list_to_binary(create_multibulk(['SET', foo, bar]))),
-
      ?_assertEqual(<<"*3\r\n$3\r\nSET\r\n$3\r\nfoo\r\n$3\r\n123\r\n">>,
                    list_to_binary(create_multibulk(['SET', foo, 123]))),
 
-     ?_assertThrow({cannot_store_floats, 123.5},
-                   list_to_binary(create_multibulk(['SET', foo, 123.5])))
-    ].
+     %% Test floats by float_to_binary(Float, [short]) or io_lib_format:fwrite_g(Float)
+     ?_assertEqual(<<"*3\r\n$3\r\nSET\r\n$3\r\nfoo\r\n$5\r\n123.5\r\n">>,
+                   list_to_binary(create_multibulk(['SET', foo, 123.5]))),
+     ?_assertEqual(<<"*3\r\n$3\r\nSET\r\n$3\r\nfoo\r\n$17\r\n3.141592653589793\r\n">>,
+                   list_to_binary(create_multibulk(['SET', foo, 3.141592653589793238462643383279]))),
```

Comment:
Redis just stores these as strings. To make redis store a float internally without the external module (time series), use a sorted sets, ZADD and ZSCORE. The *score* is stored as a double.

Example: ZADD k 3.141592653589793238462643383279 foo. Debug example using telnet:

```
$ telnet 127.0.0.1 6379
Trying 127.0.0.1...
Connected to 127.0.0.1.
Escape character is '^]'.
ZADD k 3.141592653589793238462643383279 foo
:1
ZSCORE k foo
$17
3.141592653589793
```

Btw, it seems (from my manual testing) to accept all the formats we have come up with, so I guess `io_lib:format("~.20g", Float)` is fine, for example.

We could add some roundtrip tests, starting with a float in Erlang, store in Redis, get it back to Erlang and check that we get the same float. But then we face another odd problem: Erlang requires a float to contain a dot, so it doesn't accept "4e100" and "4" but requires "4.0e100" and "4.0". It's not really a problem in eredis right now (since we don't support RESP3) but I guess you're using floats in you application and you want to get the values back to floats in Erlang, don't you? (I don't know what's the best approach for that. Try-catch with a fallback to inserting ".0" at the right place maybe...)

Reply:
That's a great catch. I'll try to update this PR with the alterations for ZADD and ZSCORE.

It's indeed a problem if Redis can return unparseable formats.. The large numbers aren't so relevant for my use case with timeseries metrics here, but I'll have to test it and see if Elixir adds anything to the parsing..

Reply:
Elixir handles this carefully with an `add_dot` function, https://github.com/elixir-lang/elixir/blob/3d98c56e69df86db3861f2035ede71d0cd83aa5b/lib/elixir/lib/float.ex#L182

```
:erlang.binary_to_float(add_dot(acc, dot?))
```

It's definitely still a huge issue for user-code.. and further adds support to the idea that throwing an error from `eredis` might be suitable, if only to notify users of this potential one-in-a-million issue :P If RESP3 support is in the long-term plan for this library, it could be a solution in the interim. Otherwise, we add a note in the README?

RE: ZADD and ZSCORE, as right now there seems to be no infrastructure in place for round-trip tests to Redis, I couldn't easily add this :) The only tests for this PR remain the ones exercising the `create_multibulk` function.